gaia-fast-list.js

```diff
@@ -264,12 +329,20 @@ Internal.prototype = {
     header.classList.add('gfl-header');
     section.appendChild(header);
     section.appendChild(background);
+    section.classList.add('gfl-section');
 
     return section;
   },
 
   populateItem: function(el, i) {
-    poplar.populate(el, this.getRecordAt(i));
+    // debug('populate item', el, i);
+    var record = this.getRecordAt(i);
+    if (!poplar.populate(el, record)) {
+      debug('not a poplar element');
+      var replacement = this.fastList.createItem();
+      this.fastList.replaceChild(replacement, el);
+      this.populateItem(replacement, i);
+    }
```

Comment:
Poplar now tells us if populate was successful or not by returning a boolean. If unsuccessful we assume that the element passed was not a poplar element and must be replaced with a new poplar element that we can template.